Extra Window: added Auto Resize and Click Through features

Blu3train opened the pull request:
Added the auto resize window height and click through mode.

The width should remain unchanged and can only be changed with the mouse, while the height adjusts to the exact number of lines of the displayed text.

In the right click menu there are the two items "Auto resize windows height" and "Click Through".
The state of the AutoResize selection is stored in the Textractor.ini file.

Added shortcut CTRL+SHIFT+T to switch the "Click Through" status.
If "Click Through" is active and you want to deactivate it, you have to give focus to the main textractor window and then press CTRL+SHIFT+T and the ExtraWindow becomes clickable again.

Blu3train commented:
Found a bug that when the text came exactly to the width of the window, sometimes the height did not fit correctly.
Slightly reduced the width of the boundingRect to calculate the required height.

mspykerez commented:
Can you make it so that the 'Click Through' will only effect left-mouse clicks and not right-mouse clicks?

Blu3train commented:
> Can you make it so that the 'Click Through' will only effect left-mouse clicks and not right-mouse clicks?

Not that I know. 
The `setAttribute(Qt::WA_TransparentForMouseEvents);` has no greater granularity.
If it were possible I would not have implemented the shortcut CTRL+SHIFT+T.

Artikash commented:
Looks pretty good. I think you had some other improvements to this extension, would you mind pushing them as well? Would like to review them all.

Blu3train commented:
I have added many other things. I try to integrate your corrections to my latest version and use them for the newly added parts and then I do a PR.
I hope to succeed, because my knowledge of C++ is very little, but I try.

Blu3train commented:
Added the following menu items:

- Hide text: hides the text by leaving a slight shadow of the window 
- Hide text mouse over: hides the text when the mouse hovers over it
- Size lock: size lock only of the window (previously also position lock)
- Position Lock: position lock of the window
- Centered text: if checked, it displays the text centered horizontally
- Show original text after translation: changes the display sequence of the source text and the translated text

Also made the following changes:
- Fixed bug that when selecting the font does not propose the current size correctly.
- Saves the last dimension of the Outline to use it as the default dimension when it is reactivated.
- Message history limited to the last 20 messages to limit memory usage
